### PR TITLE
Configure codecommit group and add examples

### PR DESCRIPTION
**Pull request by turkenf**
### Description of your changes

This PR configures resources in the codecommit group:

- [aws_codecommit_repository](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/codecommit_repository)
- [aws_codecommit_trigger](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/codecommit_trigger)
- [aws_codecommit_approval_rule_template_association](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/codecommit_approval_rule_template_association)
- [aws_codecommit_approval_rule_template](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/codecommit_approval_rule_template)

I have:

- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

`make uptest-local EXAMPLE_LIST=examples/codecommit/trigger.yaml`
`make uptest-local EXAMPLE_LIST=examples/codecommit/approvalruletemplateassociation.yaml`

### CODE REVIEW

**Comment by turkenh** [comment]
/test-examples="examples/codecommit/trigger.yaml,examples/codecommit/approvalruletemplateassociation.yaml"

**Comment by turkenh** [comment]
/test-examples="examples/codecommit/trigger.yaml"

File: apis/generate.go [hunk 1]
```diff
@@ -33,9 +33,6 @@ Copyright 2021 Upbound Inc.
 // Generate crossplane-runtime methodsets (resource.Claim, etc)
```
Comment (turkenh) [inline apis/generate.go:33]: @turkenf please remove this commit from your PR. I remember you did this to break to private dependency to unblock yourself but we should not merge it.

File: examples/codecommit/trigger.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  forProvider:
+    region: us-west-1
+---
+apiVersion: codecommit.aws.upbound.io/v1beta1
+kind: Trigger
+metadata:
+  name: example
+  annotations:
+    meta.upbound.io/example-id: codecommit/v1beta1/trigger
+    uptest.upbound.io/post-assert-hook: testhooks/delete-trigger.sh
```
Comment (turkenh) [inline examples/codecommit/trigger.yaml:30]: It looks like this is causing problems when both examples passed together.
Consider using `upjet.upbound.io/manual-intervention` annotation.

Comment (turkenh) [reply examples/codecommit/trigger.yaml:30]: We just introduced support for pre/post delete hooks with [uptest v0.2.0](https://github.com/upbound/uptest/releases/tag/v0.2.0). Please consider them to fix the problem mentioned above.

File: apis/generate.go (hunk 1 above)
Comment (turkenf) [reply apis/generate.go]: It is gone after I rebased my PR.

File: examples/codecommit/trigger.yaml (hunk 2 above)
Comment (turkenf) [reply examples/codecommit/trigger.yaml]: Done.

**Comment by turkenf** [comment]
/test-examples="examples/codecommit/trigger.yaml,examples/codecommit/approvalruletemplateassociation.yaml"

**Comment by turkenh** [comment]
/test-examples="examples/codecommit/trigger.yaml,examples/codecommit/approvalruletemplateassociation.yaml"

**Comment by turkenh** [comment]
/test-examples="examples/athena/datacatalog.yaml"

**Comment by turkenh** [comment]
/test-examples="examples/codecommit/trigger.yaml,examples/codecommit/approvalruletemplateassociation.yaml"

**Comment by turkenh** [comment]
/test-examples="examples/codecommit/trigger.yaml,examples/codecommit/approvalruletemplateassociation.yaml"

**Comment by turkenh** [comment]
/test-examples="examples/codecommit/trigger.yaml,examples/codecommit/approvalruletemplateassociation.yaml"

**Pull request merged**